docs/_data/menu-ml.yaml

```diff
@@ -1,7 +1,7 @@
 - text: Basic statistics
   url: ml-statistics.html
 - text: Data sources
-  url: ml-datasource
+  url: ml-datasource.html
```

Comment:
The orginal `url: ml-datasource` is incorrect that the generated url lose suffix `.html`
This do not matter in the offical web, since it seems that model web browser will automatically add the suffix.
However, in the locally built docs, we can not open the ml-datasource link `file:///Users/xxx/Dev/OpenSource/spark/docs/_site/ml-datasource` from page `file:///Users/xxx/Dev/OpenSource/spark/docs/_site/ml-guide.html`.